src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py

```diff
@@ -28,31 +27,30 @@ def environment_variables_format(value):
-register_cli_argument('container create', 'registry_username', arg_group='Image Registry', help='The username to log in container image registry server')
-register_cli_argument('container create', 'registry_password', arg_group='Image Registry', help='The password to log in container image registry server')
-register_cli_argument('container create', 'azure_file_volume_share_name', arg_group='Azure File Volume', help='The name of the Azure File share to be mounted as a volume')
-register_cli_argument('container create', 'azure_file_volume_account_name', arg_group='Azure File Volume', help='The name of the storage account that contains the Azure File share')
-register_cli_argument('container create', 'azure_file_volume_account_key', arg_group='Azure File Volume', help='The storage account access key used to access the Azure File share')
-register_cli_argument('container create', 'azure_file_volume_mount_path', arg_group='Azure File Volume', validator=validate_volume_mount_path, help='The path within the container where the volume should be mounted. Must not contain colon (:).')
-
-register_cli_argument('container logs', 'container_name', help='The container name to tail the logs')
+def load_arguments(self, _):
+    with self.argument_context('container') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help="The name of the container group", id_part='name')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+
+    with self.argument_context('container create') as c:
+        c.argument('location', arg_type=get_location_type(self.cli_ctx), validator=get_default_location_from_resource_group)
+        c.argument('image', help='The container image name')
+        c.argument('cpu', type=int, help='The required number of CPU cores of the containers')
+        c.argument('memory', type=float, help='The required memory of the containers in GB')
+        c.argument('os_type', arg_type=get_enum_type(OperatingSystemTypes), help='The OS type of the containers')
+        c.argument('ip_address', arg_type=get_enum_type(IP_ADDRESS_TYPES), help='The IP address type of the container group')
+        c.argument('ports', type=int, nargs='+', default=[80], help='The ports to open')
+        c.argument('restart_policy', arg_type=get_enum_type(ContainerGroupRestartPolicy), help='Restart policy for all containers within the container group')
+        c.argument('command_line', help='The command line to run when the container is started, e.g. \'/bin/bash -c myscript.sh\'')
+        c.argument('environment_variables', nargs='+', options_list=['--environment-variables', '-e'], type=_environment_variables_type, help='A list of environment variable for the container. Space separated values in \'key=value\' format.')
+        c.argument('registry_login_server', arg_group='Image Registry', help='The container image registry login server')
```

Comment:
I would suggest breaking this big block up like so:
```Python
with self.argument_context('container create') as c:
   c.argument(...)
   c.argument(...)

with self.argument_context('container create', arg_group='Image Registry') as c:
   c.argument(...)
   c.argument(...)

with self.argument_context('container create', arg_group='Azure File Volume') as c:
   c.argument(...)
   c.argument(...)
```

In that way it keeps the arguments in the "arg_group" grouped visually together and avoids repeatedly copying the arg_group to each command.

Reply:
done